src/golibmc.go

```diff
@@ -921,14 +1325,69 @@ func (client *Client) Stats() (map[string](map[string]string), error) {
 
 // Quit will close the sockets to each memcached server
 func (client *Client) Quit() error {
-	client.lock()
-	defer client.unlock()
+	client.lk.Lock()
 
-	errCode := C.client_quit(client._imp)
-	C.client_destroy_broadcast_result(client._imp)
+	close(client.openerCh)
```

Comment:
may close twice,  check ```closed = true``` first?

Reply:
That's right. Fixed.
https://github.com/douban/libmc/pull/57/commits/27ed97518881b2119b5bb2fcce70932df308359b